app/scripts/popup-core.js

```diff
@@ -21,6 +26,13 @@ function initializePopup ({ container, connectionStream }, cb) {
     (cb) => connectToAccountManager(connectionStream, cb),
     (accountManager, cb) => launchMetamaskUi({ container, accountManager }, cb),
   ], cb)
+  // firefox/ubuntu workaround, input password issue
+  // open extension in browser
+  const browser = detect()
+  if (browser && browser.name === 'firefox' && browser.os === 'Linux' && window.METAMASK_UI_TYPE !== ENVIRONMENT_TYPE_NOTIFICATION &&
```

Comment:
We could shorten this line by breaking the conditions out into their parts e.g.

```js
const isLinuxFirefox = browser && browser.name === 'firefox' && browser.os === 'Linux'
const isPopup = // ...
if (isLinuxFirefox && isPopup) {
```

Reply:
Also we should probably add more detail to the comment above, maybe a link to the GitHub issue